containers/Dockerfile.buildenv

```diff
@@ -51,11 +51,11 @@ RUN apt-get update -y \
 # Install jemalloc
 RUN apt-get update -y \
     && apt-get install -y libc++-dev libc++1 libc++abi-dev \
-    && apt-get update -y \
```

Comment:
@nilsleiffischer If I've correctly understood you, this `apt-get update` was unnecessary, so I've removed it as well as the extraneous update I had added myself.

Reply:
@nilsdeppe says this should be left as it was needed to get things to build correctly

Reply:
@nilsdeppe Fair enough. In that case I would like to update the preceding comment to make that more clear, because to me "Update is needed to get libc++ correctly" sounded like a statement about the first "update" in the list of commands here.

Do you remember what went wrong? If you don't have specific memories, then I suggest replacing that comment with something like, "The second `apt-get update` is needed to successfully install the dependencies" or something similarly vague... not tremendously useful but hopefully harder to misunderstand?

I welcome any input you might have on clarifying this! 🙂

Reply:
I don't remember exactly what went wrong. I think it failed to find `libc++-dev`, etc. Given that this works it seems fine to leave out, but I'm always scared of changes in these things because a subtle bug or issue usually ends up sucking up a lot of my time :) I would suggest `The second apt-get update is to ensure that anything that depends on libc++-dev is properly found. This was an issue on older versions of Ubuntu but might be fixed in the package manager now. To minimize changes, we are leaving the update call in for now.` Is that clear? Feel free to make any clarifying adjustments :)

Reply:
Yes, I think this is a good clarity improvement, and will help prevent some confusion should someone like me need to edit the Dockerfile in the future. Thanks! 🙂